src/app.ts

```diff
@@ -0,0 +1,373 @@
+        }
+
+        if (page) {
+            const pageHTML = page.render();
+            this.mainHTML.appendChild(pageHTML);
+            /*if (page instanceof MainPage) {
+                //viewButtonAddClick();
+                //cartButtonAddClick();
+            }
+            if (page instanceof CartPage) {
+                const buttonsItemRemove: NodeList = document.querySelectorAll('.cart__item-remove');
+                buttonsItemRemove.forEach((button) => {
+                    const id: number = parseInt((button as HTMLButtonElement).value);
+                    button.addEventListener('click', () => {
+                        removeItemFromCart(id);
+                        //                        this.renderNewPage(pageId);
+                    });
+                });
+            }*/
+        }
+    }
+
+    constructor() {
+        this.header = Header();
+        this.footer = Footer();
+    }
+
+    run() {
+        App.container.prepend(this.header);
+        loadLocalStorage();        
+        if (window.location.hash === '' || window.location.hash === '#') {
+            App.renderNewPage('');
+        } else {
+            const hash = window.location.hash.slice(1).split('/')[0];
+            const currentShoe = window.location.hash.slice(1).split('/')[1];
+            App.renderNewPage(hash);
+        }
+
+        App.container.appendChild(this.footer);
+        this.handleRouting();
+        //viewButtonAddClick();
+        //cartButtonAddClick();
+
+        /*
+        const headerCartImg = document.querySelector('.header__cart-img');
+        headerCartImg?.addEventListener('click', () => {
+            console.log(arrCart);
+        });*/
+    }
+}
+
+/* Local Storage */
+
+//save parameter in localStorage
+function saveLocalStorage() {
+    console.log('Save', arrCart);
+    localStorage.setItem('arrCart', JSON.stringify(arrCart));
+    localStorage.setItem('perPage', getPerPage().toString());
+    localStorage.setItem('currPage', getCurrPage().toString());
+    localStorage.setItem('mainLayout', mainLayout);
+}
+
+//window.addEventListener('beforeunload', saveLocalStorage);
+//window.addEventListener('load', loadLocalStorage);
+/* ------------------------- */
+
+// getHeader() {
+//     this.header = document.getElementById('header') as HTMLElement;
+//     return this.header;
+// }
+
+// const header: HTMLElement = document.getElementById('header') as HTMLElement;
+// //const main: HTMLElement = document.getElementById('main') as HTMLElement;
+// const footer: HTMLElement = document.getElementById('footer') as HTMLElement;
+
+// run() {
+
+//     header.innerHTML = headerTemplate();
+//     footer.innerHTML = footerTemplate();
+
+// }
+
+export {
+    App,
+    saveLocalStorage,
+    removeItemFromCart,
+    updateHeader,
+    addItemToCart,
+    buyNow,
+    getArrCart,
+    findInCart,
+    arrCart,
+    getMainLayout,
+    setMainLayout,
+    clearCart
+};
```

Comment:
This file is a way too big! You should split your logic into different files.
If there is a 100 or so lines of code in your file - it's very bad practice.
Your code should be as lean and transparent as possible, so other people can maintain your app if needed and shouldn't struggle with understanding it.
For me it's a complete mess there.

Reply:
resolved